### PR TITLE
Checkmate UI scaffold

### PR DESCRIPTION
**Pull request by ayazhafiz**
Very basic scaffold of the UI. Supports uploading a checkmate file to inspect and generating a trace graph for it. You can also click on nodes to show a type in the graph repr.

I will set up a CI wf to publish this to a static site for use later on.

### CODE REVIEW

File: crates/compiler/checkmate/www/src/components/Common/Variable.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  raw,
+}: VariableElProps &
+  Pick<ComponentProps<"div">, "children"> & {
+    desc: TypeDescriptor | undefined;
+  }): JSX.Element {
+  const { bg } = contentStyles(desc);
+  const varHeader =
+    !nested || raw ? (
+      <span
+        className={clsx(
```
Comment (rtfeldman) [inline crates/compiler/checkmate/www/src/components/Common/Variable.tsx:57]: TIL about `clsx`

File: crates/compiler/checkmate/www/src/components/Common/Variable.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        onClick={(e) => {
+          e.stopPropagation();
+          onClick?.(variable);
+        }}
+      >
+        {variable}
+      </span>
+    ) : (
+      <></>
+    );
```
Comment (rtfeldman) [inline crates/compiler/checkmate/www/src/components/Common/Variable.tsx:70]: I've never actually code reviewed JSX in my life until now, and I have to say this is amusing out of context:

```
    ) : (
      <></>
    );
```

Comment (ayazhafiz) [reply crates/compiler/checkmate/www/src/components/Common/Variable.tsx:70]: yes I do not love it lol

**Review by rtfeldman — APPROVED** [review]
This is without a doubt the best JSX code I have ever reviewed in my entire life! 🏆 

Seriously though, I'm very excited to see dedicated tooling to making it easier to debug the compiler. 👍 

Thanks for making the investment! 🎉

**Pull request merged**